admin_scripts/Dell/firmware_version_check.sh

```diff
@@ -37,6 +37,13 @@ do
   printf "%-5s"
 
   # Pull hardware model number through dmidecode, and filter out "Poweredge" out of model name.
-  timeout 15 ssh $i dmidecode -s system-product-name | awk '{ print $2 }'
+  model=$(timeout 15 ssh $i dmidecode -s system-product-name | awk '{ print $2 }')
+
+  # If the above command times out, print that it timed out
+  if [ -z "$model" ]; then
+    printf "Timed Out \n"
+  else
+    printf "%s \n" "$model"
+  fi
```

Comment:
I do not think you want the spaces before the `\n` characters:

```suggestion
  if [ -z "$model" ]; then
    printf "Timed Out\n"
  else
    printf "%s\n" "$model"
  fi
```

I think that is adding an ending space on each line which you probably do not want.

Reply:
Oh good catch, this was probably out of habit for legibility, but should be cut out.